### DISCUSSION TITLE
hardhat handling multiple tx limits?

### DISCUSSION THREAD

**Discussion by mattador1** [NomicFoundation/hardhat#4662] [Troubleshooting]
Hi, is there any limit how much hardhat can take tx submitted to process/mine.
I have mulithread app and around 15 transaction are submitted every minute at the same time (which is 5 different method calls but each is called 3x with diff parameters) , hardhat is on automine mode. Is there any limits to its performance, as when I have single thread and submitting (and waiting for one transaction to complete) all is ok, but when I increase to 15 tx/minute I get: Error: VM Exception while processing transaction: reverted with an unrecognized custom error (return data: 0x32d1517a) on some transactions

**Comment by fvictorio** [comment]
Hi, there isn't a limit like that, but with automining enabled we process one transaction at a time while the others wait. Perhaps the ones "waiting" are not following a FIFO order and that causes issues. Is it possible that that's what's going on?

**Comment by mattador1** [comment]
I don't think order of transactions should matter, when tx is then executed, is atomic, also have in mind those method calls just emit events, there is no modifying of state
